File: src/main/javascript/onboarding.js
while (true) {
    var count = parseInt(readline()); // The number of current enemy ships within range
    var enemies = [];
    for (var i = 0; i < count; i++) {
        var inputs = readline().split(' ');
        enemies.push({name: inputs[0], dist: parseInt(inputs[1])});
    }

    enemies.sort(function(a, b) {return a.dist - b.dist;});
    print(enemies[0].name);
}

File: src/main/scala/Onboarding.scala
import math._
import scala.util._

object Player {
    def main(args: Array[String]) {
        while(true) {
            val count = readInt // The number of current enemy ships within range
            val enemies = 1 to count map { _ => readLine split " " match {case Array(x, y) => (x, y toInt)} }
            println(enemies.minBy(enemy => enemy._2)._1) // The name of the most threatening enemy (HotDroid is just one example)
        }
    }
}
